### PR TITLE
Imprtance Updated to Work with sorting

### PR DESCRIPTION
**Pull request by liuxess**

### CODE REVIEW

File: src/main/java/com/kustudents/issuetracker/service/ImportanceService.java [hunk 1]
```diff
@@ -16,25 +16,56 @@
...
+    }
+
     public List<Importance> getImportanceList(){
-        return importanceRepo.findAll(Sort.by("id"));
+        return importanceRepo.findAll(Sort.by("sortOrder"));
     }
 
     public Importance getImportanceByID(Long ID){
-        return importanceRepo.getOne(ID);
+        return importanceRepo.findById(ID).orElseThrow();
```
Comment (ValdasPiekus) [inline src/main/java/com/kustudents/issuetracker/service/ImportanceService.java]: Throw what?

File: src/main/java/com/kustudents/issuetracker/service/ImportanceService.java [hunk 2]
```diff
@@ -16,25 +16,56 @@
...
+        importances.removeIf(importance -> importance.getSortOrder().compareTo(importanceToDelete.getSortOrder())<=0);//Removes all Importances that have a lower Importance order
+        importances.forEach(importance -> {importance.setSortOrder(importance.getSortOrder()-1);});//decreases the leftover sort orders by 1
+        importanceRepo.delete(importanceToDelete);
         importanceRepo.saveAll(importances);
     }
 
-    public Importance createImportance(Importance importance){
-        return importanceRepo.save(importance);
+    private void updateImportance(Importance importance){
+        Importance importanceToUpdate = importanceRepo.findById(importance.getId()).orElseThrow();
```
Comment (ValdasPiekus) [inline src/main/java/com/kustudents/issuetracker/service/ImportanceService.java]: Throw what?

File: src/main/java/com/kustudents/issuetracker/service/ImportanceService.java [hunk 3]
```diff
@@ -16,25 +16,56 @@
...
+        importances.forEach((importance) ->{
+            updateImportance(importance);
+        });
     }
 
-    public void rewriteImportances(List<Importance> importances){
-        importanceRepo.deleteAllInBatch();
+    public void deleteImportanceById(Long id){
+        if(importanceRepo.count() == 1){
+            //TODO: Implement Exception
```
Comment (ValdasPiekus) [inline src/main/java/com/kustudents/issuetracker/service/ImportanceService.java]: throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "There can not be less than 1 importance"); - should work

**Review by ValdasPiekus — CHANGES_REQUESTED** [review]

File: src/main/java/com/kustudents/issuetracker/service/ImportanceService.java (hunk 1 above)
Comment (liuxess) [reply src/main/java/com/kustudents/issuetracker/service/ImportanceService.java]: Implemented

File: src/main/java/com/kustudents/issuetracker/service/ImportanceService.java (hunk 3 above)
Comment (liuxess) [reply src/main/java/com/kustudents/issuetracker/service/ImportanceService.java]: Implemented

File: src/main/java/com/kustudents/issuetracker/service/ImportanceService.java (hunk 2 above)
Comment (liuxess) [reply src/main/java/com/kustudents/issuetracker/service/ImportanceService.java]: implemented

**Review by ValdasPiekus — APPROVED** [review]

**Pull request merged**